refactor(asserts): extract common utilities and improve retry logic

matt-metivier opened the pull request:
- Add common utilities for client validation and retry logic
- Refactor alert config and disabled alert config resources to use shared utilities
- Improve retry logic with exponential backoff and jitter
- Remove code duplication across asserts resources
- Update test destroy functions with better retry logic
- Remove stress test skip conditions as retry logic is now more robust

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/grafana/terraform-provider-grafana?pullRequest=2336) <br/>All committers have signed the CLA.

matt-metivier committed:
refactor(asserts): extract common utilities and improve retry logic

- Add common utilities for client validation and retry logic
- Refactor alert config and disabled alert config resources to use shared utilities
- Improve retry logic with exponential backoff and jitter
- Remove code duplication across asserts resources
- Update test destroy functions with better retry logic
- Remove stress test skip conditions as retry logic is now more robust
- Update asserts client dependency for compatibility with new API features

matt-metivier committed:
test: add stress test flag to control eventual consistency tests

- Add CheckStressTestsEnabled function to testutils package
- Update stress tests to require TF_ACC_STRESS environment variable
- Stress tests will now be skipped unless explicitly enabled
- This prevents flaky stress test failures in CI while still allowing
  developers to run them when needed for testing eventual consistency

matt-metivier committed:
Merge branch 'main' into asserts-refactor-common-utilities

arramos84 reviewed (approved):
This LGTM, tested the provider as well. Seems to be working as intended.
